src/System.Text.Json/tests/Serialization/CustomConverterTests.Object.cs

```diff
@@ -259,5 +259,198 @@ private class TestClassWithFieldsHavingCustomConverter
             public int IntValue { get; set; }
             public string Message { get; set; }
         }
+
+        /// <summary>
+        /// A converter that converts a JSON "true" and "false" tokens to a System.Bool,
+        /// a JSON number to a System.Double, and a JSON string to either System.String (if not a date)
+        /// or System.DateTimeOffset or System.Date depending on whether the date has an offset.
+        /// </summary>
+        private class ObjectToPrimitiveConverter : JsonConverter<object>
+        {
+            public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.True)
+                {
+                    return true;
+                }
+
+                if (reader.TokenType == JsonTokenType.False)
+                {
+                    return false;
+                }
+
+                if (reader.TokenType == JsonTokenType.Number)
+                {
+                    return reader.GetDouble();
+                }
+
+                if (reader.TokenType == JsonTokenType.String)
```

Comment:
What about Guid?

Reply:
I could add it... Maybe cross-reference with Json.NET?

Reply:
Now basing semantics on Json.NET which doesn't do anything with Guid.